### PR TITLE
Adapt UNet2D for supre-resolution

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adapts the `UNet2DConditionModel` for Stable diffusion 2 upscaler.

### CODE REVIEW

File: src/diffusers/models/attention.py [hunk 1]
```diff
@@ -387,14 +389,17 @@ def __init__(
         activation_fn: str = "geglu",
         num_embeds_ada_norm: Optional[int] = None,
         attention_bias: bool = False,
+        do_self_attention: bool = True,
```
Comment (patrickvonplaten) [inline src/diffusers/models/attention.py:392]: just in terms of naming, the first attn is actually called "CrossAttention", so should we maybe just call it as follows:

```suggestion
        disable_first_cross_attention: bool = True,
```

File: src/diffusers/models/unet_2d_condition.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def __init__(
             "DownBlock2D",
         ),
         up_block_types: Tuple[str] = ("UpBlock2D", "CrossAttnUpBlock2D", "CrossAttnUpBlock2D", "CrossAttnUpBlock2D"),
+        do_self_attention: Tuple[bool] = (True, True, True, True),
```
Comment (patrickvonplaten) [inline src/diffusers/models/unet_2d_condition.py:101]: only some layers use "double cross attention"?

Comment (patrickvonplaten) [reply src/diffusers/models/unet_2d_condition.py:101]: Actually ok for me as is - don't think we have to add new blocks for this.
Just in terms of naming I'd call it "double_cross_attention"

Comment (patrickvonplaten) [reply src/diffusers/models/unet_2d_condition.py:101]:
```suggestion
        double_cross_attention: Union[bool, Tuple[bool]] = False,
```

File: src/diffusers/models/unet_2d_condition.py [hunk 3]
```diff
@@ -109,6 +110,7 @@ def __init__(
         attention_head_dim: Union[int, Tuple[int]] = 8,
         dual_cross_attention: bool = False,
         use_linear_projection: bool = False,
+        num_classes: Optional[int] = None,
```
Comment (patrickvonplaten) [inline src/diffusers/models/unet_2d_condition.py:113]:
```suggestion
        num_classes_embeds: Optional[int] = None,
```

**Review by patrickvonplaten — COMMENTED** [review]
PR actually looks good to me as is - would just rename a couple of things

File: src/diffusers/models/unet_2d_condition.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/models/unet_2d_condition.py:101]: As @pcuenca mentioned, `disable_self_attention: Union[bool, Tuple[bool]] = False` might be good as well. But double negatives are always a bit hard to grasp. 

Maybe:
`only_cross_attention` could also be a good config name. @pcuenca @anton-l @patil-suraj wdyt?

File: src/diffusers/models/unet_2d_condition.py (hunk 3 above)
Comment (patrickvonplaten) [reply src/diffusers/models/unet_2d_condition.py:113]: Here "embeds" is important IMO as this defines more or less exactly how many class embedding vectors there are

File: src/diffusers/models/unet_2d_condition.py (hunk 2 above)
Comment (pcuenca) [reply src/diffusers/models/unet_2d_condition.py:101]: I like `only_cross_attention` !

**Review by pcuenca — APPROVED** [review]
Looks good!

**Pull request merged**